File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-ird-5-4/Notional.cpp
// Notional.cpp 
#include "Notional.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

Notional::Notional(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //notionalStepScheduleNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* notionalStepScheduleNode = xmlNode->FirstChildElement("notionalStepSchedule");

   if(notionalStepScheduleNode){notionalStepScheduleIsNull_ = false;}
   else{notionalStepScheduleIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- notionalStepScheduleNode , address : " << notionalStepScheduleNode << std::endl;
   #endif
   if(notionalStepScheduleNode)
   {
      if(notionalStepScheduleNode->Attribute("href") || notionalStepScheduleNode->Attribute("id"))
      {
          if(notionalStepScheduleNode->Attribute("id"))
          {
             notionalStepScheduleIDRef_ = notionalStepScheduleNode->Attribute("id");
             notionalStepSchedule_ = boost::shared_ptr<NonNegativeAmountSchedule>(new NonNegativeAmountSchedule(notionalStepScheduleNode));
             notionalStepSchedule_->setID(notionalStepScheduleIDRef_);
             IDManager::instance().setID(notionalStepScheduleIDRef_,notionalStepSchedule_);
          }
          else if(notionalStepScheduleNode->Attribute("href")) { notionalStepScheduleIDRef_ = notionalStepScheduleNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { notionalStepSchedule_ = boost::shared_ptr<NonNegativeAmountSchedule>(new NonNegativeAmountSchedule(notionalStepScheduleNode));}
   }

   //notionalStepParametersNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* notionalStepParametersNode = xmlNode->FirstChildElement("notionalStepParameters");

   if(notionalStepParametersNode){notionalStepParametersIsNull_ = false;}
   else{notionalStepParametersIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- notionalStepParametersNode , address : " << notionalStepParametersNode << std::endl;
   #endif
   if(notionalStepParametersNode)
   {
      if(notionalStepParametersNode->Attribute("href") || notionalStepParametersNode->Attribute("id"))
      {
          if(notionalStepParametersNode->Attribute("id"))
          {
             notionalStepParametersIDRef_ = notionalStepParametersNode->Attribute("id");
             notionalStepParameters_ = boost::shared_ptr<NotionalStepRule>(new NotionalStepRule(notionalStepParametersNode));
             notionalStepParameters_->setID(notionalStepParametersIDRef_);
             IDManager::instance().setID(notionalStepParametersIDRef_,notionalStepParameters_);
          }
          else if(notionalStepParametersNode->Attribute("href")) { notionalStepParametersIDRef_ = notionalStepParametersNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { notionalStepParameters_ = boost::shared_ptr<NotionalStepRule>(new NotionalStepRule(notionalStepParametersNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<NonNegativeAmountSchedule> Notional::getNotionalStepSchedule()
{
   if(!this->notionalStepScheduleIsNull_){
        if(notionalStepScheduleIDRef_ != ""){
             return boost::shared_static_cast<NonNegativeAmountSchedule>(IDManager::instance().getID(notionalStepScheduleIDRef_));
        }else{
             return this->notionalStepSchedule_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<NonNegativeAmountSchedule>();
   }
}
boost::shared_ptr<NotionalStepRule> Notional::getNotionalStepParameters()
{
   if(!this->notionalStepParametersIsNull_){
        if(notionalStepParametersIDRef_ != ""){
             return boost::shared_static_cast<NotionalStepRule>(IDManager::instance().getID(notionalStepParametersIDRef_));
        }else{
             return this->notionalStepParameters_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<NotionalStepRule>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-ird-5-4/Notional.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class Notional : ISerialized
    {
        public Notional(XmlNode xmlNode)
        {
            XmlNode notionalStepScheduleNode = xmlNode.SelectSingleNode("notionalStepSchedule");
            
            if (notionalStepScheduleNode != null)
            {
                if (notionalStepScheduleNode.Attributes["href"] != null || notionalStepScheduleNode.Attributes["id"] != null) 
                {
                    if (notionalStepScheduleNode.Attributes["id"] != null) 
                    {
                        notionalStepScheduleIDRef_ = notionalStepScheduleNode.Attributes["id"].Value;
                        NonNegativeAmountSchedule ob = new NonNegativeAmountSchedule(notionalStepScheduleNode);
                        IDManager.SetID(notionalStepScheduleIDRef_, ob);
                    }
                    else if (notionalStepScheduleNode.Attributes["href"] != null)
                    {
                        notionalStepScheduleIDRef_ = notionalStepScheduleNode.Attributes["href"].Value;
                    }
                    else
                    {
                        notionalStepSchedule_ = new NonNegativeAmountSchedule(notionalStepScheduleNode);
                    }
                }
                else
                {
                    notionalStepSchedule_ = new NonNegativeAmountSchedule(notionalStepScheduleNode);
                }
            }
            
        
            XmlNode notionalStepParametersNode = xmlNode.SelectSingleNode("notionalStepParameters");
            
            if (notionalStepParametersNode != null)
            {
                if (notionalStepParametersNode.Attributes["href"] != null || notionalStepParametersNode.Attributes["id"] != null) 
                {
                    if (notionalStepParametersNode.Attributes["id"] != null) 
                    {
                        notionalStepParametersIDRef_ = notionalStepParametersNode.Attributes["id"].Value;
                        NotionalStepRule ob = new NotionalStepRule(notionalStepParametersNode);
                        IDManager.SetID(notionalStepParametersIDRef_, ob);
                    }
                    else if (notionalStepParametersNode.Attributes["href"] != null)
                    {
                        notionalStepParametersIDRef_ = notionalStepParametersNode.Attributes["href"].Value;
                    }
                    else
                    {
                        notionalStepParameters_ = new NotionalStepRule(notionalStepParametersNode);
                    }
                }
                else
                {
                    notionalStepParameters_ = new NotionalStepRule(notionalStepParametersNode);
                }
            }
            
        
        }
        
    
        #region NotionalStepSchedule_
        private NonNegativeAmountSchedule notionalStepSchedule_;
        public NonNegativeAmountSchedule NotionalStepSchedule_
        {
            get
            {
                if (this.notionalStepSchedule_ != null)
                {
                    return this.notionalStepSchedule_; 
                }
                else if (this.notionalStepScheduleIDRef_ != null)
                {
                    notionalStepSchedule_ = IDManager.getID(notionalStepScheduleIDRef_) as NonNegativeAmountSchedule;
                    return this.notionalStepSchedule_; 
                }
                else
                {
                      return this.notionalStepSchedule_; 
                }
            }
            set
            {
                if (this.notionalStepSchedule_ != value)
                {
                    this.notionalStepSchedule_ = value;
                }
            }
        }
        #endregion
        
        public string notionalStepScheduleIDRef_ { get; set; }
        #region NotionalStepParameters_
        private NotionalStepRule notionalStepParameters_;
        public NotionalStepRule NotionalStepParameters_
        {
            get
            {
                if (this.notionalStepParameters_ != null)
                {
                    return this.notionalStepParameters_; 
                }
                else if (this.notionalStepParametersIDRef_ != null)
                {
                    notionalStepParameters_ = IDManager.getID(notionalStepParametersIDRef_) as NotionalStepRule;
                    return this.notionalStepParameters_; 
                }
                else
                {
                      return this.notionalStepParameters_; 
                }
            }
            set
            {
                if (this.notionalStepParameters_ != value)
                {
                    this.notionalStepParameters_ = value;
                }
            }
        }
        #endregion
        
        public string notionalStepParametersIDRef_ { get; set; }
        
    
        
    
    }
    
}

